src/collection_validator.py

```diff
@@ -0,0 +1,104 @@
+import pystac
+from utils import get_tif_metadata
+from datetime import datetime, timedelta
+
+
+class CollectionValidator:
```

Comment:
The separation between this and the spec validator is good in terms of logic, But I think this class should be named "collection", look at the functions:  load_items, create_collection, and create_items, and I think they are good that way.

Reply:
Done